twoopstracker/twitterclient/twitter_client.py

```diff
@@ -48,7 +50,16 @@ def on_delete(self, status_id, user_id):
         mark_tweet_as_deleted.delay(status_id)
 
     def on_data(self, data):
+        if (time.time() - self.start_time) > (
+            settings.TWOOPTRACKER_STREAM_LISTENER_INTERVAL * 60
+        ) - 5:  # disconnect 5 seconds before the restart
```

Comment:
My understanding is this method is called only when we received something from Twitter. If this is correct, then this is dangerous because it's possible not to receive something for a few minutes and by the time we receive something, it could well be that we've already tried to start another stream.

Reply:
I agree @kilemensi 
I've refactored the code differently now.